usr/share/rear/output/default/94_grub2_rescue.sh

```diff
@@ -11,7 +11,10 @@ is_true "$GRUB_RESCUE" || return
 # (grub-probe or grub2-probe only exist in GRUB 2)
 # in particular do not run this script when GRUB Legacy is used
 # (for GRUB Legacy output/default/94_grub_rescue.sh is run):
-type -p grub-probe >&2 || type -p grub2-probe >&2 || { LogPrint "Skipping GRUB_RESCUE setup for GRUB 2 (no GRUB 2 found)." ; return ; }
+if [[ ! $( type -p grub-probe ) && ! $( type -p grub2-probe ) ]]; then
```

Comment:
Now I start to understand. 
You are talking about "cleanness" when debugging enabled :-) that explains a lot.
I was looking more on log, when debugging was disabled.